### PR TITLE
Transfers: added oidc_account to poller #5457

### LINKED ISSUE #5457: conveyor-poller does not attempt to use tokens for authentication with FTS
We're not able to get conveyor-poller to use tokens to poll FTS. Submitter works fine.

Had a cursory look at the code, and can't seem to find where [oidc_account ](https://github.com/rucio/rucio/blob/master/lib/rucio/transfertool/fts3.py#L634) is set in FTS3TransferTool for the polling flow. For submitter, it seems like it's set [here ](https://github.com/rucio/rucio/blob/release-1.28/lib/rucio/transfertool/fts3.py#L714)(called by [next_transfers_to_submit](https://github.com/rucio/rucio/blob/release-1.28/lib/rucio/daemons/conveyor/submitter.py#L65)).

Without this variable being set, polling falls back to x509 authentication with FTS. This is obviously not compatible with our goal to have an end-to-end token-only workflow.

### PR DESCRIPTION
**Pull request by robbarnsley**

### CODE REVIEW

File: lib/rucio/daemons/conveyor/poller.py [hunk 1]
```diff
@@ -95,7 +96,9 @@ def run_once(fts_bulk, db_bulk, older_than, activity_shares, multi_vo, timeout,
                 if TRANSFER_TOOL == 'globus':
                     transfertool_obj = GlobusTransferTool(external_host=None)
                 else:
-                    transfertool_obj = FTS3Transfertool(external_host=external_host, vo=vo)
+                    if oidc_account:
+                        account = InternalAccount(oidc_account, vo=vo)
+                    transfertool_obj = FTS3Transfertool(external_host=external_host, vo=vo, oidc_account=account)
```
Comment (rcarpa) [inline lib/rucio/daemons/conveyor/poller.py:102]: the `account` variable may be accessed before assignment. The tests will fail for this PR

Comment (rcarpa) [reply lib/rucio/daemons/conveyor/poller.py:102]: Other than the comment above, looks good to me

**Comment by robbarnsley** [comment]
As I mentioned in #5649, I really think it would be prudent to be more judicious about type annotations.

I'm now failing checks for adding an argument to an existing function that doesn't have annotations. I don't really know much about this function, or what it accepts apart from the argument i've added, so i'd have to trawl the code to make sure I fully understood it before I felt comfortable adding annotations for the rest of the arguments.

**Comment by bari12** [comment]
I explained this a bit in my other comment, but I get your point for this too; I would recommend to just add the annotation for the newly added parameter. The test will still fail, but we can ignore it in that case.

**Review by rcarpa — APPROVED** [review]

**Pull request merged**